rules/windows/privilege_escalation_dns_serverlevelplugindll.toml

```diff
@@ -0,0 +1,55 @@
+[metadata]
+creation_date = "2024/05/29"
+integration = ["endpoint"]
+maturity = "production"
+updated_date = "2024/05/29"
+
+[rule]
+author = ["Elastic"]
+description = """
+Identifies unusual DLLs loaded by the DNS Server process, potentially indicating the abuse of the ServerLevelPluginDll
+functionality. This can lead to privilege escalation and remote code execution with SYSTEM privileges.
+"""
+from = "now-9m"
+index = ["logs-endpoint.events.library-*"]
+language = "eql"
+license = "Elastic License v2"
+name = "Potential DNS Server Privilege Escalation via ServerLevelPluginDll"
+references = [
+    "https://cube0x0.github.io/Pocing-Beyond-DA/",
+    "https://adsecurity.org/?p=4064",
+    "https://github.com/gtworek/PSBits/tree/master/ServerLevelPluginDll"
+]
+risk_score = 47
+rule_id = "5d676480-9655-4507-adc6-4eec311efff8"
+severity = "medium"
+tags = [
+    "Domain: Endpoint",
+    "OS: Windows",
+    "Use Case: Threat Detection",
+    "Tactic: Privilege Escalation",
+    "Data Source: Elastic Defend"
+]
+timestamp_override = "event.ingested"
+type = "eql"
+
+query = '''
+library where host.os.type == "windows" and
+  process.name : "dns.exe" and
+  not dll.code_signature.subject_name : "Microsoft*"
```

Comment:
We could just look for unsigned DLLs or `\Device\Mup` too, but this way doesn't look too noisy

Reply:
looking for unsigned/untrustec-signed DLLs should be acceptable for most cases.